src/view_maybe/view_maybe.h

```diff
@@ -8,130 +8,91 @@
     constexpr T*       begin() noexcept { return data(); }
     constexpr const T* begin() const noexcept { return data(); }
-    constexpr T*       end() noexcept {
-        if (data())
-            return data() + 1;
-        else
-            return data();
-    }
+    constexpr T*       end() noexcept { return data() + size(); }
+    constexpr const T* end() const noexcept { return data() + size(); }
 
-    constexpr const T* end() const noexcept {
-        if (data())
-            return data() + 1;
-        else
-            return data();
-    }
-
-    constexpr std::ptrdiff_t size() noexcept {
-        if (value_.get())
-            return 1;
-        else
-            return 0;
-    }
+    constexpr std::ptrdiff_t size() noexcept { return bool(value_.get()); }
 
     constexpr T* data() noexcept {
-        if (value_.get())
-            return std::addressof(*value_.get());
+        Maybe& m = value_.get();
+        if (m)
+            return std::addressof(*m);
         else
-            return 0;
+            return nullptr;
     }
-
     constexpr const T* data() const noexcept {
-        if (value_.get())
-            return std::addressof(*value_.get());
+        Maybe& m = value_.get();
+        if (m)
+            return std::addressof(*m);
         else
-            return 0;
+            return nullptr;
     }
 };
 
 template <Nullable Maybe>
 class ref_maybe_view
     : public std::experimental::ranges::view_interface<ref_maybe_view<Maybe>> {
-    std::remove_reference_t<Maybe>* value_;
-    using R = std::remove_reference_t<decltype(**value_)>;
+    using T = std::remove_reference_t<ranges::iter_reference_t<Maybe>>;
+
+    Maybe* value_ = nullptr;
 
   public:
     constexpr ref_maybe_view() = default;
-    constexpr ref_maybe_view(ref_maybe_view const&) = default;
+    constexpr explicit ref_maybe_view(Maybe& maybe) noexcept
+        : value_(std::addressof(maybe)) {}
 
-    constexpr explicit ref_maybe_view(Maybe& maybe) : value_(std::addressof(maybe)) {}
-
-    constexpr R*       begin() noexcept { return data(); }
-    constexpr const R* begin() const noexcept { return data(); }
-    constexpr R*       end() noexcept {
-        if (data())
-            return data() + 1;
-        else
-            return data();
-    }
-
-    constexpr const R* end() const noexcept {
-        if (data())
-            return data() + 1;
-        else
-            return data();
-    }
+    constexpr T*       begin() noexcept { return data(); }
+    constexpr const T* begin() const noexcept { return data(); }
+    constexpr T*       end() noexcept { return data() + size(); }
+    constexpr const T* end() const noexcept { return data() + size(); }
 
-    constexpr std::ptrdiff_t size() noexcept {
-        if (**value_)
-            return 1;
-        else
-            return 0;
-    }
+    constexpr std::ptrdiff_t size() const noexcept { return bool(*value_); }
```

Comment:
I noticed the lack of `const` here, but not in the other class??!? Wat?

Reply:
Edit error at some point. Much of this was Saturday afternoon at San Diego.

Reply:
My confusion isn't about *how* these became non-`const`, but about how I managed to fix one and didn't check the other.